contracts/governance/Vesting/fouryear/FourYearVestingLogic.sol

```diff
@@ -108,13 +114,26 @@ contract FourYearVestingLogic is IFourYearVesting, FourYearVestingStorage, Appro
         emit VotesDelegated(msg.sender, _delegatee);
     }
 
+    /**
```

Comment:
it feels like duplicating code for VestingLogic and FourYearVestingLogic
because we are going to redeploy both anyway
could you take a chance and restructure it a bit - extract all common functionality into a separate contract to inherit from both contracts

Reply:
no changes
pls comment this decision to keep code duplication

Reply:
We keep the code duplication for now.

Reply:
could you pls give a reason for that? 
it doesn't look like a big change - just a simple restructuring - extracting common code into an abstract contract and inheritance. 
else we need to remember to change both on every change.

Reply:
same as for the other vesting changes. please, separate them from this PR.

Reply:
Vesting logic and four year vesting logic has been reverted in this [commit](https://github.com/DistributedCollective/Sovryn-smart-contracts/pull/447/commits/21dd284dc55c0b0355bb165ce2e24dbf154e32e6)